core/meta/src/TClass.cxx

```diff
@@ -4199,19 +4199,15 @@ TMethod *TClass::GetMethod(const char *method, const char *params,
 /// Find a method with decl id in this class or its bases.
 
 TMethod* TClass::FindClassOrBaseMethodWithId(DeclId_t declId) {
-   TFunction *f = GetMethodList()->Get(declId);
-   if (f) return (TMethod*)f;
+   if (auto method = GetMethodList()->Get(declId))
+      return reinterpret_cast<TMethod *>(method);
 
-   TBaseClass *base;
-   TIter       next(GetListOfBases());
-   while ((base = (TBaseClass *) next())) {
-      TClass *clBase = base->GetClassPointer();
-      if (clBase) {
-         f = clBase->FindClassOrBaseMethodWithId(declId);
-         if (f) return (TMethod*)f;
-      }
-   }
-   return 0;
+   for (auto item : *GetListOfBases())
```

Comment:
One hesitation I have with this pattern is that the iterator cast is moved one level down. i.e.
```
while ((item = reinterpret_cast<TBaseClass *>(next()))) {
   if (auto base = item->GetClassPointer())
```
vs
```
for (auto item : *GetListOfBases())
   if (auto base = reinterpret_cast<TBaseClass *>(item)->GetClassPointer())
```
or more often.
```
for (auto item : *GetListOfBases()) {
   auto base = reinterpret_cast<TBaseClass *>(item);
   if (base->GetClassPointer())
```

I wonder if we can have something like

```
for (auto item : ContaineeIs<TBaseClass*>(GetListOfBases()))
   if (auto base = item->GetClassPointer())
```

Reply:
Let's not sidestep a 5 line commit into a lenghty discussion on code patterns. If you disagree that the code is more readable after the changes, just let me know and I'll close the PR.